### PR TITLE
Add higher res icons to meson install script

### PR DESCRIPTION
**Pull request by weirddan455**
# Description
The meson installer was only installing icons up to 32X32.  This led to bad quality icons when displayed at larger sizes.  Include larger sizes up to 256X256 (everything in `contrib/icons/png`).

## Related issues
#3169

# Manual testing
```
meson setup -Dbuildtype=debug -Dunit_tests=disabled build/debug
meson compile -C build/debug
meson install -C build/debug --destdir fakeinstall
```

This creates a "fake install" directory inside the build directory.  Icons are located in `fakeinstall/usr/local/share/icons/`.

Manually copied these icons to my `~/.local/share/icons` folder.  Tested with KDE's "large icon task switcher".  Requires deleting `~/.cache/icon-cache.kcache` and logging out and back into KDE to refresh the session after adding or removing icon files.

Before:

icon

After:

icons-fixed

# Checklist

_Please tick the items as you have addressed them. Don't remove items; leave the ones that are not applicable unchecked._

I have:

- [x] followed the project's [contributing guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/master/CONTRIBUTING.md) and [code of conduct](https://github.com/dosbox-staging/dosbox-staging/blob/master/CODE_OF_CONDUCT.md).
- [x] performed a self-review of my code.
- [ ] commented on the particularly hard-to-understand areas of my code.
- [ ] split my work into well-defined, bisectable commits, and I [named my commits well](https://github.com/dosbox-staging/dosbox-staging/blob/main/CONTRIBUTING.md#commit-messages).
- [x] applied the appropriate labels (bug, enhancement, refactoring, documentation, etc.)
- [x] [checked](https://github.com/dosbox-staging/dosbox-staging/blob/main/scripts/compile_commits.sh) that all my commits can be built.
- [ ] confirmed that my code does not cause performance regressions (e.g., by running the Quake benchmark).
- [ ] added unit tests where applicable to prove the correctness of my code and to avoid future regressions.
- [ ] made corresponding changes to the documentation or the website according to the [documentation guidelines](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md).
- [ ] [locally verified](https://github.com/dosbox-staging/dosbox-staging/blob/main/DOCUMENTATION.md#previewing-documentation-changes-locally) my website or documentation changes.

### CODE REVIEW

File: contrib/icons/meson.build [hunk 1]
```diff
@@ -22,6 +22,21 @@ install_data(
...
+install_data(
+    'png' / 'icon_96.png',
+    rename: 'dosbox-staging.png',
+    install_dir: icons_hicolor / '96x96' / 'apps',
+)
+install_data(
+    'png' / 'icon_256.png',
+    rename: 'dosbox-staging.png',
+    install_dir: icons_hicolor / '256x256' / 'apps',
+)
```
Comment (shermp) [inline contrib/icons/meson.build:39]: Looking at the meson documentation, you should be able to do something like
```meson
install_data(
    ['png/icon_32.png', 'png/icon_48.png', 'png/icon_96.png', 'png/icon_256.png'],
    rename: ['32x32/apps/dosbox-staging.png', '48x48/apps/dosbox-staging.png', '96x96/apps/dosbox-staging.png', '256x256/apps/dosbox-staging.png'],
    install_dir: icons_hicolor
)
```
Or perhaps a loop could be used?

Comment (johnnovak) [reply contrib/icons/meson.build:39]: There was some looping previously but I removed it. My bad 😄 

Any reason not to add 512 and 1024? That would add just a few Ks. Or are sizes about 256 not supported on Linux?

Comment (weirddan455) [reply contrib/icons/meson.build:39]: There are no 512 and 1024 sized pngs checked into the repo.  I suppose I could generate them and check them in with this PR.  I don't think the icon Makefile ever gets run except when we manually run it.  This meson.build just copies the images.

1024X1024 didn't work on my system but that's going to depend on desktop environment and possibly display settings as well.  As I'm on a 1080p monitor, it's not surprising that it didn't try to load such a large icon.

Comment (weirddan455) [reply contrib/icons/meson.build:39]: >  Yes, and that's explained in the README in contrib/icon. I also provided you with steps a few days ago on how to add the 512 & 1024 icons to the repo, remember? 😄

Yes, I remember and I was just looking over that README as well.  I thought someone had mentioned the icon Makefile was used somewhere in our normal build system (or in CI) but I'm not seeing where if that's the case.

Either way, I'll add the larger sizes real quick.

**Review by kcgen — APPROVED** [review]

File: contrib/icons/meson.build (hunk 1 above)
Comment (weirddan455) [reply contrib/icons/meson.build]: 512X512  and 1024X1024 icons have been generated + added to the Makefile.  Also changed the meson.build to use a loop + a dict.  Looks like a cleaner solution to me  but I had to look up syntax for Meson loops so I'm doing this on the fly :wink:

Comment (johnnovak) [reply contrib/icons/meson.build]: > I thought someone had mentioned the icon Makefile was used somewhere in our normal build system (or in CI) but I'm not seeing where if that's the case.

I was that, `make install` is used for the Linux packaging stuff, and that's it.

File: contrib/icons/Makefile [hunk 2]
```diff
@@ -40,9 +40,11 @@ png:
 	# Rendering this from the SVG optimised for 24x24px looks good enough.
 	rsvg-convert -h  22 svg/dosbox-staging-24.svg > $(PNG_DIR)/icon_22.png
 
-	rsvg-convert -h  48 svg/dosbox-staging.svg    > $(PNG_DIR)/icon_48.png
-	rsvg-convert -h  96 svg/dosbox-staging.svg    > $(PNG_DIR)/icon_96.png
-	rsvg-convert -h 256 svg/dosbox-staging.svg    > $(PNG_DIR)/icon_256.png
+	rsvg-convert -h   48 svg/dosbox-staging.svg   > $(PNG_DIR)/icon_48.png
+	rsvg-convert -h   96 svg/dosbox-staging.svg   > $(PNG_DIR)/icon_96.png
+	rsvg-convert -h  256 svg/dosbox-staging.svg   > $(PNG_DIR)/icon_256.png
```
Comment (johnnovak) [inline contrib/icons/Makefile:45]: How about adding 128x128 as well?

File: contrib/icons/meson.build [hunk 3]
```diff
@@ -1,27 +1,27 @@
 icons_hicolor = data_dir / 'icons' / 'hicolor'
 
+png_files = {
+    'icon_16.png': '16x16',
+    'icon_22.png': '22x22',
+    'icon_24.png': '24x24',
+    'icon_32.png': '32x32',
+    'icon_48.png': '48x48',
+    'icon_96.png': '96x96',
```
Comment (johnnovak) [inline contrib/icons/meson.build:9]: 128x128 is missing.

File: contrib/icons/meson.build [hunk 4]
```diff
@@ -1,27 +1,27 @@
...
-    install_dir: icons_hicolor / '32x32' / 'apps',
-)
+
+foreach name, size : png_files
+    install_data(
+        'png' / name,
+        rename: 'dosbox-staging.png',
+        install_dir: icons_hicolor / size / 'apps',
+    )
+endforeach
```
Comment (kcgen) [inline contrib/icons/meson.build:24]: Very nice :metal:

File: contrib/icons/Makefile (hunk 2 above)
Comment (weirddan455) [reply contrib/icons/Makefile:46]: Done

File: contrib/icons/meson.build (hunk 3 above)
Comment (weirddan455) [reply contrib/icons/meson.build:9]: Added

**Review by johnnovak — APPROVED** [review]
Nice one @weirddan455! 🎉 Time to claim your bounty 😎 💸

**Pull request merged**